app/controllers/subscriptions_controller.rb

```diff
@@ -0,0 +1,35 @@
+class SubscriptionsController < ApplicationController
+  before_action :set_user, :set_locale
+  skip_authorization_check
+
+  def edit
+  end
+
+  def update
+    @user.update!(subscriptions_params)
+    redirect_to edit_subscriptions_path(token: @user.subscriptions_token),
+                notice: t("flash.actions.save_changes.notice")
+  end
+
+  private
+
+    def set_user
+      @user = User.find_by(subscriptions_token: params[:token])
+    end
+
+    def subscriptions_params
+      attributes = [:email_on_comment, :email_on_comment_reply, :email_on_direct_message,
+                    :email_digest, :newsletter]
+      params.require(:user).permit(*attributes)
+    end
```

Comment:
Using an additional method would make customization easier:

```suggestion
    def subscriptions_params
      params.require(:user).permit(allowed_params)
    end
    
    def allowed_params
       [:email_on_comment, :email_on_comment_reply, :email_on_direct_message, :email_digest, :newsletter]
     end
```

This way the custom code could say:

```
  alias_method :consul_allowed_params, :allowed_params

  def allowed_params
    consul_allowed_params + [:my_custom_parameter]
  end
```